tests/network/services.go

```diff
@@ -0,0 +1,159 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"fmt"
+	"strconv"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	batchv1 "k8s.io/api/batch/v1"
+	v12 "k8s.io/api/core/v1"
+	v13 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+)
+
+var _ = SIGDescribe("Services", func() {
+	var virtClient kubecli.KubevirtClient
+
+	runTCPClientExpectingHelloWorldFromServer := func(host, port, namespace string) *batchv1.Job {
+		job := tests.NewHelloWorldJob(host, port)
+		job, err := virtClient.BatchV1().Jobs(namespace).Create(job)
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		return job
+	}
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		Expect(err).NotTo(HaveOccurred(), "Should successfully initialize an API client")
+	})
+
+	Context("bridge interface binding", func() {
+		var inboundVMI *v1.VirtualMachineInstance
+		var serviceName string
+
+		const (
+			selectorLabelKey = "expose"
+			selectorLabelValue = "me"
+			servicePort = 1500
+		)
+
+		BeforeEach(func() {
+			// inboundVMI expects implicitly to be added to the pod network
+			inboundVMI = tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
+			inboundVMI.Labels = map[string]string{selectorLabelKey: selectorLabelValue}
+			inboundVMI.Spec.Subdomain = "myvmi"
+			inboundVMI.Spec.Hostname = "my-subdomain"
+			// Remove the masquerade interface to use the default bridge one
+			inboundVMI.Spec.Domain.Devices.Interfaces = nil
+			inboundVMI.Spec.Networks = nil
+
+			_, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(inboundVMI)
```

Comment:
1. I think we are missing the delete of the VMI.
2. The fact that you are now using "BeforeEach" when creating the VMI is a major change which is hidden here. It merits special note in the commit message with the cost and reasoning.
We do not have a good solution to this resource sharing between tests problem, the only thing that comes to mind is wrapping ginkgo with a template function, but that needs a dedicated work and is out of this scope.

Reply:
I have on https://github.com/kubevirt/kubevirt/pull/4035/commits/980b3b35f20a023a5869af29338b3debb5f8f93e#diff-9a20ab1179984538941b03db7cae82e9R67 ... I'll move it to the first commit.

Reply:
Same, not squashed.

Reply:
deleting the VMI is being done below, on https://github.com/kubevirt/kubevirt/pull/4035/commits/1c7cf19c069f5ffc1f8faf9eda644ecad58f3d93#diff-9a20ab1179984538941b03db7cae82e9R86 .

Only when I add the masquerade part, on a follow-up commit do I extract that as a function.

Reply:
Ohh now I see it. Sorry.
Please mark "Done" and close this.

Reply:
Done.